frame/support/src/traits.rs

```diff
@@ -1533,11 +1533,15 @@ pub trait OnGenesis {
 	fn on_genesis() {}
 }
 
+pub const ON_RUNTIME_UPGRADE_PREFIX: &[u8] = b"__ON_RUNTIME_UPGRADE__";
+
 /// The runtime upgrade trait.
 ///
 /// Implementing this lets you express what should happen when the runtime upgrades,
 /// and changes may need to occur to your module.
 pub trait OnRuntimeUpgrade {
+	const ID: &'static str;
```

Comment:
I could feature-gate this as well.. but then I think the `try-runtime` feature will have to propagate to almost all pallets? if so, don't want that. But will give it a try.